LayoutTests/http/tests/permissions/dedicated-worker-permission-query.html

```diff
@@ -0,0 +1,64 @@
+<!DOCTYPE html>
+<html>
+<head>
+<script src="/resources/js-test-pre.js"></script>
+<script src="/resources/notifications-test-pre.js"></script>
+</head>
+<body>
+<script>
+
+description("This test checks that Permissions::query() works for dedicated workers")
+
+jsTestIsAsync = true;
+
+var expectedData = null;
+var receivedPostMessageResponse = true;
+
+var worker = new Worker('dedicated-worker-permission-query.js');
+worker.onmessage = function(message) {
+    window.message = message;
+    shouldBeEqualToString("message.data", expectedData);
+    receivedPostMessageResponse = true;
+}
+
+async function defaultTest()
+{
+    receivedPostMessageResponse = false;
+    expectedData = "prompt";
+    worker.postMessage(1);
+    await shouldBecomeEqual("receivedPostMessageResponse", "true");
+}
+
+async function grantTest() 
+{   
+    receivedPostMessageResponse = false;
+    expectedData = "granted";
+    testRunner.setGeolocationPermission(true);
+    worker.postMessage(2);
+    await shouldBecomeEqual("receivedPostMessageResponse", "true");
+}
+
+async function denyTest() 
+{
+    receivedPostMessageResponse = false;
+    // FIXME: Current behavior changes denied to prompt in WebPageProxy. Eventually, we want
```

Comment:
I think the FIXME might be added in code (e.g. WebPageProxy::queryPermission) instead of in test.